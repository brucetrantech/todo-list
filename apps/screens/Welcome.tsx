import { Image, StyleSheet, Text, View } from "react-native";
import contents from "@/commons/contents";
import images from "@/commons/images";
import screens from "@/commons/screens";
import themes from "@/commons/themes";
import { YTButton, YTLayout } from "@/cores";

type WelcomeProps = {
	navigation: any;
}

export default function Welcome({ navigation }: WelcomeProps) {

	const onNavigate = () => navigation.navigate(screens.SIGNIN);

	return (
		<YTLayout>
			<View style={styles.view}>
				<Image source={images.welcome} resizeMode="contain" />
				<Text style={styles.title}>{contents.WELCOME_TITLE}</Text>
				<Text style={styles.description}>{contents.LOREMS}</Text>
				<YTButton title={contents.STARTED} onPress={onNavigate} variant="contained" />
			</View>
		</YTLayout>
	)
}

const styles = StyleSheet.create({
	view: {
		flex: 1,
		justifyContent: 'center',
	},
	title: {
		...themes.font.mediumBold,
		color: themes.color.dark,
		paddingTop: 40,
		paddingBottom: 20,
		textAlign: 'center',
	},
	description: {
		...themes.font.normal,
		textAlign: 'center',
		paddingBottom: 80,
	},
})