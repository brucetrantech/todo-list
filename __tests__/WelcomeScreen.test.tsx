import React from 'react';
import renderer from 'react-test-renderer';
import Welcome from '@/screens/Welcome';

test('r[Snapshot] Welcome screen renders correctly', () => {
  const tree = renderer.create(<Welcome navigation={null} />).toJSON();
  expect(tree).toMatchSnapshot();
});