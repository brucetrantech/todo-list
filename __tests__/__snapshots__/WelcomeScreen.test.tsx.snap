// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r[Snapshot] Welcome screen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../apps/assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "testUri": "../../../apps/assets/images/welcome.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "Poppins-Medium",
              "fontSize": 14,
              "fontWeight": "800",
              "paddingBottom": 20,
              "paddingTop": 40,
              "textAlign": "center",
            }
          }
        >
          Get things done with todo - Your Tasks
        </Text>
        <Text
          style={
            {
              "color": "#3E3E3E",
              "fontFamily": "Poppins-Medium",
              "fontSize": 12,
              "fontWeight": "500",
              "paddingBottom": 80,
              "textAlign": "center",
            }
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing. Maxime, tempore! Animi nemo aut atque, deleniti nihil dolorem repellendus.
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#55847A",
              "borderColor": "#55847A",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 42,
              "minWidth": 42,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                },
                {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Get started
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../apps/assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "testUri": "../../../apps/assets/images/welcome.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "Poppins-Medium",
              "fontSize": 14,
              "fontWeight": "800",
              "paddingBottom": 20,
              "paddingTop": 40,
              "textAlign": "center",
            }
          }
        >
          Get things done with todo - Your Tasks
        </Text>
        <Text
          style={
            {
              "color": "#3E3E3E",
              "fontFamily": "Poppins-Medium",
              "fontSize": 12,
              "fontWeight": "500",
              "paddingBottom": 80,
              "textAlign": "center",
            }
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing. Maxime, tempore! Animi nemo aut atque, deleniti nihil dolorem repellendus.
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#55847A",
              "borderColor": "#55847A",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 42,
              "minWidth": 42,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                },
                {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Get started
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`renders snapshot correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../apps/assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "testUri": "../../../apps/assets/images/welcome.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "Poppins-Medium",
              "fontSize": 14,
              "fontWeight": "800",
              "paddingBottom": 20,
              "paddingTop": 40,
              "textAlign": "center",
            }
          }
        >
          Get things done with todo - Your Tasks
        </Text>
        <Text
          style={
            {
              "color": "#3E3E3E",
              "fontFamily": "Poppins-Medium",
              "fontSize": 12,
              "fontWeight": "500",
              "paddingBottom": 80,
              "textAlign": "center",
            }
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing. Maxime, tempore! Animi nemo aut atque, deleniti nihil dolorem repellendus.
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#55847A",
              "borderColor": "#55847A",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 42,
              "minWidth": 42,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Poppins-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                },
                {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Get started
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
